### PR TITLE
Two new tests: dirDepthUpload and dirDepthDownload

### PR DESCRIPTION
**Pull request by jnfrmarks**
Added two new tests:

test_dirDepthUpload.py - creates files locally, syncs to server; focus is on directory depth and structure

test_dirDepthDownload.py - creates files via on server, syncs to locally, focus is on directory depth and structure

@jvillafanez @SergioBertolinSG @PVince81 @nickvergessen

### CODE REVIEW

File: etc/smashbox.conf.template [hunk 1]
```diff
@@ -132,3 +132,9 @@ reset_server_log = False
 # used in a test run to allow for more users sharing at time
 #
 share_sets = 1
+
+# properties for defining directory depth and directory layout
+#
+dir_depth = 5
```
Comment (nickvergessen) [inline etc/smashbox.conf.template:143]: I think stuff like this should not be in the config, but on the test itself?

**Comment by nickvergessen** [comment]
Needs to be rebased, sorry

**Comment by jnfrmarks** [comment]
What's the rebase magic?

File: etc/smashbox.conf.template (hunk 1 above)
Comment (jnfrmarks) [inline etc/smashbox.conf.template:143]: I get that but would rather not make the change now ;)  The way smashbox is currently designed, the properties file is used for defining test properties.  We can create an issue to refactor the property handling in general if you want.

**Comment by nickvergessen** [comment]
```
git checkout master
git pull origin master
git checkout dir-depth
git rebase origin/master
```

**Comment by jnfrmarks** [comment]
Not really sure what I did ;)  but ... 

@nickvergessen

File: etc/smashbox.conf.template [hunk 2]
```diff
@@ -136,3 +136,10 @@ share_sets = 1
 #
 # scp port to be used in scp commands, used primarily when copying over the server log file
 scp_port=22
+
+#
+# properties for defining directory depth and directory layout
+#
+dir_depth = 5
```
Comment (nickvergessen) [inline etc/smashbox.conf.template:143]: are these values even used, I checked out your branch, without doing the config changes and they still pass.
The reason is, that the values from the testsets are put into the config on `run_test` (`bin/smash`):

``` py
      def run_test(ts,i=0):
         if ts:
            for opt,val in zip(ts.keys(),ts.values()):
               setattr(config,opt,val)
               ...
```

So I think we can leave the config change out.

**Comment by nickvergessen** [comment]
Hmm okay, looking at the commit list something went wrong on the rebase.
Mind doing a new PR?

**Comment by jnfrmarks** [comment]
@nickvergessen 
How do I do the new PR?  Just compare the two branches again?

**Comment by jnfrmarks** [comment]
Closed in favor of 76

**Pull request closed without merging**